File: src/main/java/dominio/metodologias/Cuantificador.java
package dominio.metodologias;

import dominio.empresas.Empresa;

import javax.persistence.*;
import java.time.Year;

@Entity
@Table(name = "cuantificadores")
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
public abstract class Cuantificador {

	@Id 
	@GeneratedValue
	protected Long id;
	
	public abstract int evaluarEn(Empresa empresa, Year anio);
	
}

File: src/main/java/dominio/metodologias/Cuantificador.kt
package dominio.metodologias

import dominio.empresas.Empresa

import javax.persistence.*
import java.time.Year

@Entity
@Table(name = "cuantificadores")
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
abstract class Cuantificador {

    @Id
    @GeneratedValue
    protected var id: Long? = null

    abstract fun evaluarEn(empresa: Empresa, anio: Year): Int

}
